docs/fundamentals/app-lifecycle.md

```diff
@@ -0,0 +1,440 @@
+                    {
+                        System.Diagnostics.Debug.WriteLine($"Lifecycle event: {eventName}{(type == null ? string.Empty : $" ({type})")}");
+                    }
+                });
+
+            return builder.Build();
+        }
+    }
+}
+```
+
+### Retrieve the Window object
+
+Native code on each platform can retrieve the app's `Window` object from native lifecycle events, with the `GetWindow` extension method:
+
+```csharp
+using Microsoft.Maui;
+using Microsoft.Maui.Controls.Hosting;
+using Microsoft.Maui.LifecycleEvents;
+
+namespace NativeLifecycleDemo
+{
+    public static class MauiProgram
+    {
+        public static MauiApp CreateMauiApp()
+        {
+            var builder = MauiApp.CreateBuilder();
+            builder
+                .UseMauiApp<App>()
+                .ConfigureLifecycleEvents(events =>
+                {
+#if WINDOWS
+                    events.AddWindows(windows => windows
+                            .OnClosed((window, args) =>
+                            {
+                                IWindow appWindow = window.GetWindow();
+                            }));
+#endif
+                });
+
+            return builder.Build();
+        }
+    }
+}
+```
+
+## Custom lifecycle events
+
+While .NET MAUI defines delegates that are invoked in response to native platform lifecycle events being raised, it only exposes a common set of native platform lifecycle events. However, it also includes a mechanism, typically for library authors, that enables apps to be notified when additional native platform lifecycle events are raised. The process for accomplishing this is as follows:
+
+- Register an event handler for a native lifecycle event that isn't exposed by .NET MAUI.
+- In the event handler for the native lifecycle event, retrieve the `ILifecycleEventService` instance and call its `InvokeEvents` method, specifying the native event name as its argument.
+
+Then, apps that want to receive notification of the native lifecycle event should modify the `CreateMauiApp` method of their `MauiProgram` class to call the `ConfigureLifecycleEvents` method on the `MauiAppBuilder` object. Then, on the `ILifecycleBuilder` object, call the `AddEvent` method and specify the native event name and the `Action` that will be invoked when the native event is raised.
+
+### Example
+
+The WinUI 3 [`Window.SizeChanged`](xref:Microsoft.UI.Xaml.Window.SizeChanged) event occurs when the native app window has first rendered, or has changed its rendering size. .NET MAUI doesn't expose this native event as a lifecycle event. However, apps can receive notification when this native event is raised by using the following approach:
+
+- Register an event handler for the [`Window.SizeChanged`](xref:Microsoft.UI.Xaml.Window.SizeChanged) native lifecycle event:
+
+    ```csharp
+    #if WINDOWS
+                MauiWinUIApplication.Current.MainWindow.SizeChanged += OnSizeChanged;
```

Comment:
Instead of using the static `MainWindow` here can we combine this with the example above? 

Show users that they can do something like this

```C#
windows
				.OnWindowCreated(window =>
				{
					window.SizeChanged += OnSizeChanged;
				})
```